heartbeat/service/synthetics_service.go

```diff
@@ -0,0 +1,277 @@
+			}
+
+		case <-reloadPushTicker.C:
+			for locationKey, serviceLocation := range serviceLocations {
+				service.servicePushWait.Add(1)
+				// first we need to do at start, and then ticker will take over
+				go service.pushConfigsToSyntheticsService(locationKey, serviceLocation, output)
+			}
+			reloadPushTicker.Stop()
+		}
+	}
+}
+
+func (service *SyntheticService) getSyntheticServiceManifest() (config.ServiceManifest, error) {
+	serviceCfg := service.config.Service
+	var err error
+
+	if serviceCfg.Username == "" {
+		err = errors.New("synthetic service username is required for authentication")
+	}
+
+	if serviceCfg.Password == "" {
+		err = errors.New("synthetic service password is required for authentication")
+	}
+
+	if serviceCfg.ManifestURL == "" {
+		err = errors.New("synthetic service manifest url is required")
+	}
+
+	if err != nil {
+		return config.ServiceManifest{}, err
+	}
+
+	req, err := http.NewRequest("GET", serviceCfg.ManifestURL, nil)
+
+	resp, err := Client.Do(req)
+
+	if err != nil {
+		return config.ServiceManifest{}, err
+	}
+
+	serviceManifest := config.ServiceManifest{}
+
+	read, err := ioutil.ReadAll(resp.Body)
+
+	err = json.Unmarshal(read, &serviceManifest)
+
+	return serviceManifest, err
+
+}
+
+func (service *SyntheticService) validateMonitorsSchedule() error {
+	for _, m := range service.config.Monitors {
+		monitorFields, _ := stdfields.ConfigToStdMonitorFields(m)
+		monitorSchedule, _ := schedule.ParseSchedule(monitorFields.ScheduleStr)
+		if monitorSchedule.Seconds() < 60 {
+			return errors.New("schedule can't be less than 1 minute while using synthetics service")
+		}
+	}
+	return nil
+}
+
+func (service *SyntheticService) pushConfigsToSyntheticsService(locationKey string, serviceLocation config.ServiceLocation, output Output) {
+	defer service.servicePushWait.Done()
+
+	payload := SyntheticServicePayload{Output: output}
+
+	addToPayload := func(monCfg *common.Config) {
+		monitorFields, _ := stdfields.ConfigToStdMonitorFields(monCfg)
+		if locationInServiceLocation(locationKey, monitorFields.ServiceLocations) {
+			target := map[string]interface{}{}
+			err := monCfg.Unpack(target)
+			if err != nil {
+				logp.Info("error unpacking monitor plugin config")
+				return
+			}
+			payload.Monitors = append(payload.Monitors, target)
+		}
+	}
+	monitorsById := map[string]*common.Config{}
+	for _, monCfg := range service.config.Monitors {
+		monitorFields, _ := stdfields.ConfigToStdMonitorFields(monCfg)
+		monitorsById[monitorFields.ID] = monCfg
+	}
+
+	if service.config.ConfigMonitors.Enabled() {
+		for monId, monitor := range service.serviceRunnerFactory.GetMonitorsById() {
+			monitorsById[monId] = monitor
+		}
+	}
+
+	for _, monitor := range monitorsById {
+		addToPayload(monitor)
+	}
+
+	if len(payload.Monitors) == 0 {
+		logp.Info("No monitor found for service: %s, to push configuration.", serviceLocation.Geo.Name)
+		return
+	}
+
+	serviceCfg := service.config.Service
```

Comment:
defining this variable so far from where it's used is confusing. I'd either move it just before `SetBasicAuth` or not use a variable at all.